### ISSUE TITLE
`quotes`: allowTemplateLiterals: "avoidEscape" does not work as expected

### ISSUE THREAD

**Issue by dzek69** [eslint-stylistic/eslint-stylistic#1007]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Hello,
I recently updated this plugin and I noticed something I wanted - an extended allowTemplateLiterals option.

So I tweaked my rules to `double` and these options
```javascript
{
  avoidEscape: false,
  allowTemplateLiterals: "avoidEscape",
  ignoreStringLiterals: false,
}
```

And I was surprised to see that it autofixed my ``["n:1","i:2","u:","b:1","b:","l:"]`` into `"[\"n:1\",\"i:2\",\"u:\",\"b:1\",\"b:\",\"l:\"]"`

The docs may not be up to date (no examples on `allowTemplateLiterals: "avoidEscape",` and examples presents now deprecated boolean values for `allowTemplateLiterals`), so I'm really not sure what's the desired output, but my assumptions were:

CORRECT:
```javascript
const a = "string";
const b = `This is a "string"`;
```

INCORRECT:
```javascript
const a = 'string'; // should be autofixed to "string"
const b = `This is a string`; // should be autofixed to "This is a string"
const c = 'This is a "string"';
```

Now the last part is tricky. By just reading the config I know what's allowed, but not what's preferred. Ideally I'd prefer to see `This is a "string"`, but let's say I'd accept `"This is a \"string\"";` because it's technically correct.

I'd like to ask for a bugfix and maybe considering upgrading the config even more, to specify preferences too? 🤔 

### Reproduction

https://stackblitz.com/edit/github-w3htnq8d-eaxkydfm?file=src%2Findex.ts

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by 9romise** [comment]
Sorry for the delayed response. The `allowTemplateLiterals: "avoidEscape"` option was introduced in #634 and currently must be used together with `avoidEscape: true`.

Here are some related historical changes:
- https://github.com/eslint/eslint/issues/5147
- https://github.com/eslint/eslint/commit/fb3c2eb7ce6a07507e48b0fe1986dc15ebc847b2

Due to historical reasons, the configuration of this rule can indeed be a bit tricky. If you have a better design in mind, we’d love to hear your thoughts!
